api/src/telemetry/lib/init-telemetry.ts

```diff
@@ -0,0 +1,38 @@
+import { getCache } from '../../cache.js';
+import { useEnv } from '../../env.js';
+import { scheduleSynchronizedJob } from '../../utils/schedule.js';
+import { toBoolean } from '../../utils/to-boolean.js';
+import { track } from './track.js';
+
+/**
+ * Exported to be able to test the anonymous callback function
+ */
+export const jobCallback = () => {
+	track();
+};
+
+/**
+ * Initialize the telemetry tracking. Will generate a report on start, and set a schedule to report
+ * every 6 hours
+ *
+ * @returns Whether or not telemetry has been initialized
+ */
+export const initTelemetry = async () => {
+	const env = useEnv();
+
+	if (toBoolean(env['TELEMETRY']) === false) return false;
+
+	scheduleSynchronizedJob('telemetry', '0 */6 * * *', jobCallback);
+
+	const { lockCache } = getCache();
+
+	if (!(await lockCache.get('telemetry-lock'))) {
+		await lockCache.set('telemetry-lock', true, 30000);
```

Comment:
Since the get and set operations aren't done atomically, this could theoretically lead to more than one instance acquiring the lock when one instance gets the lock and another instance gets the lock before the first instance could set it. But since `lockCache` is used in a similar fashion elsewhere, I guess it's fine? 🤷‍♂️

Reply:
You're not wrong! I think the extreme unlikeliness of multiple containers hitting it on the same fraction of a second is what has saved us in the past. The plan is to replace the whole cache setup with the Kv/LRU stuff we've prepared in `@directus/memory`, so I'm not terribly worried about over-optimizing that now in this PR 🙂

Reply:
That's what I figured, hence the non-blocking review 🙂